src/mame/drivers/evmbug.cpp

```diff
@@ -51,12 +60,32 @@ class evmbug_state : public driver_device
 	required_device<generic_terminal_device> m_terminal;
 };
 
+class tms9995bb_state : public evmbug_state
+{
+public:
+	tms9995bb_state(const machine_config &mconfig, device_type type, const char *tag)
+		: evmbug_state(mconfig, type, tag)
+	{ }
+
+	void tms9995bb(machine_config &config);
+
+protected:
+	void mem_map(address_map &map);
+};
```

Comment:
Does this actually need an extra state class?

Reply:
I *think* that's the only way of doing the different memory map, but happy to take an alternative approach if you can point me in the right direction?